community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -828,6 +823,9 @@ private Procedures setupProcedures() throws KernelException, IOException
         procedures.registerComponent( GraphDatabaseService.class,
                 (ctx) -> dependencyResolver.resolveDependency( GraphDatabaseService.class ) );
 
+        InternalProcedureLoggingService loggingService = new InternalProcedureLoggingService( log.getUserLog( Procedures.class  ) );
+        procedures.registerComponent( LoggingService.class, (ctx) -> loggingService );
```

Comment:
I assume it is public but not entirely sure. `LogProvider` is exposed from `GraphDatabaseFacadeFactory.Dependencies` interface which can be used to create `CommunityNeoServer` which is probably public.